### PR TITLE
Add PR/Issue/Feature templates

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
@Madhu-1 @ShyamsundarR PTAL .

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+1. Scroll down to '....'
+1. See error
+
+**Actual results**
+Describe what happened
+
+**Expected behavior**
+A clear and concise description of what you expected to happen.
+
+**Additional context**
```
Comment (Madhu-1) [inline .github/ISSUE_TEMPLATE/bug_report.md]: need to add what are the logs needs to be collected to report a bug

Comment (Madhu-1) [reply .github/ISSUE_TEMPLATE/bug_report.md]: i think we need below info also
what image version used 
what helm chat version used
kube version used
if requried kube templates

Comment (humblec) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Sure, adding those.

**Comment by humblec** [comment]
@Madhu-1 added, Please let me know if you think something obvious which can be added or we can improve it later based on our experience on handling issues/PRs.

**Comment by humblec** [comment]
@Madhu-1 @ShyamsundarR PTAL.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Describe what this PR does #
+
+Provide some context for the reviewer
+
+## Is there anything that requires special attention ##
```
Comment (ShyamsundarR) [inline .github/pull_request_template.md]: @humblec request addition of a section along the lines of "Provide any external context for the change", to accommodate things **like** Kubernetes links that explain why a change is required for example, or CSI spec related changes that may need to be pointed to, or generic golang related practices that may need to be pointed to.

Asking for this, as a lot of PRs assume familiarity with other eco-systems, and it would be nice to provide links to relate to them, than assume the same, as it helps those without the eco-system details to understand it better.

Hence, suggesting to provide it as an added section may make folks provide the same.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+Do you have any questions? Did you do something clever?
+
+## Related issues ##
+
+Mention any github issues relevant to this PR
```
Comment (ShyamsundarR) [inline .github/pull_request_template.md]: Without making it any longer :) would it make sense to add "Future concerns" section to this template? This is to list items that are not part of the PR and do not impact it's functionality, but are work items that can be taken up in the subsequently.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+# Describe the bug #
+
+A clear and concise description of what the bug is.
+
+# Steps to reproduce #
+
+Steps to reproduce the behavior:
+
+1. Go to '...'
```
Comment (ShyamsundarR) [inline .github/ISSUE_TEMPLATE/bug_report.md]: This list does not add value from a CSI perspective I feel. There is nothing to click/scroll etc. we should change this to something more appropriate for CSI.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+A clear and concise description of what you expected to happen.
+
+# Additional context #
+
+Add any other context about the problem here.
+
+For example:
+
+- Image/version of Ceph CSI driver
```
Comment (ShyamsundarR) [inline .github/ISSUE_TEMPLATE/bug_report.md]: I would move the version information request much ahead as a separate section in itself, as it is important to understand the working env before proceeding with possible analysis and troubleshot.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+- helm chart version
+- Kubernetes cluster version
+- Logs
```
Comment (ShyamsundarR) [inline .github/ISSUE_TEMPLATE/bug_report.md]: (future) When we have a troubleshooting/self-help guide we should link it here.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+1. See error
+
+# Actual results #
+
+Describe what happened
+
+# Expected behavior #
+
+A clear and concise description of what you expected to happen.
+
```
Comment (Madhu-1) [inline .github/ISSUE_TEMPLATE/bug_report.md]: need to add one more  section for
`Details on how to reproduce`

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 5 above)
Comment (Madhu-1) [inline .github/ISSUE_TEMPLATE/bug_report.md]: This should go as the separate Environment details.

File: .github/pull_request_template.md (hunk 2 above)
Comment (humblec) [reply .github/pull_request_template.md]: Done. Or used the same word in above.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 5 above)
Comment (humblec) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Its now.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 7 above)
Comment (humblec) [reply .github/ISSUE_TEMPLATE/bug_report.md]: `Steps to reproduce` already exist

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 6 above)
Comment (humblec) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Sure.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 5 above)
Comment (humblec) [reply .github/ISSUE_TEMPLATE/bug_report.md]: yep, its done.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 4 above)
Comment (humblec) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Added something for now. But its self explanatory , so shouldnt be needed a documentation as such.

File: .github/pull_request_template.md (hunk 3 above)
Comment (humblec) [reply .github/pull_request_template.md]: Added

**Comment by humblec** [comment]
@Madhu-1 @ShyamsundarR PTAL

**Comment by humblec** [comment]
@ShyamsundarR @Madhu-1 Any thing pending here ?

**Comment by ShyamsundarR** [comment]
Leaving a quick comment, will get to this tomorrow.

I did not intend that my text in the reviews be used as is, so I would possibly provide some text that can be used as is, or if you can reword the same @humblec it would be good.

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
+# Describe what this PR does #
+
+Provide some context for the reviewer
+
+## Is there anything that requires special attention ##
+
+Do you have any questions? Did you do something clever?
+
```
Comment (ShyamsundarR) [inline .github/pull_request_template.md]: Add a line here to remind of backward compatibility concerns, like so,

"Is the change backward compatible? Or, are there concerns around backward compatibility?"

File: .github/pull_request_template.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+need to be pointed to, or generic golang related practices
+that may need to be pointed to.
+
+## Related issues ##
+
+Mention any github issues relevant to this PR
+
+## Future concerns ##
+
+list items that are not part of the PR and do not impact it's
```
Comment (ShyamsundarR) [inline .github/pull_request_template.md]: (minor nit) Capitalize first letter of this sentence

(nit) ending should be "...but are work items that can be taken up ~~in the~~ subsequently."

File: .github/pull_request_template.md [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+## Is there anything that requires special attention ##
+
+Do you have any questions? Did you do something clever?
+
+Provide any external context for the change, if any.
+
+For example: Kubernetes links that explain why a change is
+required, or CSI spec related changes that may
+need to be pointed to, or generic golang related practices
+that may need to be pointed to.
```
Comment (ShyamsundarR) [inline .github/pull_request_template.md]: (nits) Reword slightly

"Provide any available external context for the change.

For example:
  - Kubernetes links that explain why the change is required
  - CSI spec related changes/catch-up that necessitates this patch
  - golang related practices that necessitates this change"

File: .github/pull_request_template.md (hunk 8 above)
Comment (humblec) [reply .github/pull_request_template.md]: Done.

File: .github/pull_request_template.md (hunk 9 above)
Comment (humblec) [reply .github/pull_request_template.md]: Done.

File: .github/pull_request_template.md (hunk 10 above)
Comment (humblec) [reply .github/pull_request_template.md]: Done.

**Comment by humblec** [comment]
@Madhu-1 @ShyamsundarR Addressed the comments. PTAL

File: .github/pull_request_template.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+For example:
+
+    * Kubernetes links that explain why the change is required
+    * CSI spec related changes/catch-up that necessitates this patch
+    * golang related practices that necessitates this change"
+
+## Related issues ##
+
+Mention any github issues relevant to this PR
```
Comment (Madhu-1) [inline .github/pull_request_template.md]: add only one more line for `Fixes: #issue_number`  for auto-closing the issue once PR is merged

Comment (humblec) [reply .github/pull_request_template.md]: done.
